oak-run-commons/src/test/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileNodeStoreBuilderTest.java

```diff
@@ -60,60 +87,162 @@ public void defaultSortStrategy() throws Exception {
-            when(indexerSupport.retrieveNodeStateForCheckpoint()).thenReturn(rootState);
-            List<FlatFileStore> storeList = builder.buildList(indexHelper, indexerSupport, null);
-            assertEquals(1, storeList.size());
-        } finally {
-            System.clearProperty(OAK_INDEXER_SORTED_FILE_PATH);
+    public void testBuildListSplitLZ4() throws CompositeException, IOException {
+        System.setProperty(OAK_INDEXER_USE_ZIP, "true");
+        System.setProperty(OAK_INDEXER_USE_LZ4, "true");
+        System.setProperty(IndexerConfiguration.PROP_OAK_INDEXER_MIN_SPLIT_THRESHOLD, "0");
+        LZ4Compression compression = new LZ4Compression();
+
+        File newFlatFile = getFile("complex-split.json", compression);
+        System.setProperty(OAK_INDEXER_SORTED_FILE_PATH, newFlatFile.getParentFile().getAbsolutePath());
+        
+        assertBuildList(newFlatFile.getParentFile().getAbsolutePath(), true);
+    }
+
+    public void assertBuild(String dir) throws CompositeException, IOException {
+        FlatFileNodeStoreBuilder builder = new FlatFileNodeStoreBuilder(folder.getRoot()).withNodeStateEntryTraverserFactory(
+            nodeStateEntryTraverserFactory);
+        FlatFileStore store = builder.build();
+        assertEquals(dir, store.getFlatFileStorePath());
+    }
+    
+    private File getFile(String dataFile, Compression compression) throws IOException {
+        File flatFile = new File(getClass().getClassLoader().getResource(dataFile).getFile());
+        File newFlatFile = new File(folder.getRoot(), FlatFileStoreUtils.getSortedStoreFileName(compression));
+        try (BufferedReader reader = FlatFileStoreUtils.createReader(flatFile, false);
+            BufferedWriter writer = FlatFileStoreUtils.createWriter(newFlatFile, compression)) {
+            IOUtils.copy(reader, writer);
+        }
+        return newFlatFile;
+    }
+
+    public void assertBuildList(String dir, boolean split) throws CompositeException, IOException {
+        FlatFileNodeStoreBuilder builder = new FlatFileNodeStoreBuilder(folder.getRoot()).withNodeStateEntryTraverserFactory(
+            nodeStateEntryTraverserFactory);
+        IndexHelper indexHelper = mock(IndexHelper.class);
+        when(indexHelper.getWorkDir()).thenReturn(new File(dir));
+        IndexerSupport indexerSupport = mock(IndexerSupport.class);
+        NodeState rootState = mock(NodeState.class);
+        when(indexerSupport.retrieveNodeStateForCheckpoint()).thenReturn(rootState);
+        
+        List<FlatFileStore> storeList = builder.buildList(indexHelper, indexerSupport, mockIndexDefns());
+        
+        if (split) {
+            assertEquals(new File(dir, "split").getAbsolutePath(), storeList.get(0).getFlatFileStorePath());
+            assertTrue(storeList.size() > 1);
+        } else {
+            assertEquals(dir, storeList.get(0).getFlatFileStorePath());
+            assertTrue(storeList.size() == 1);
```

Comment:
it can be simplified
```suggestion
            assertEquals(1, storeList.size());
```